src/industry_gui.cpp

```diff
@@ -831,9 +832,24 @@ class IndustryViewWindow : public Window
 		NWidgetViewport *nvp = this->GetWidget<NWidgetViewport>(WID_IV_VIEWPORT);
 		nvp->InitializeViewport(this, Industry::Get(window_number)->location.GetCenterTile(), ScaleZoomGUI(ZOOM_LVL_INDUSTRY));
 
+		int count = 0;
+		const Industry *i = Industry::Get(window_number);
+		for (size_t j = 0; j < INDUSTRY_NUM_OUTPUTS; j++) {
+			if (!IsCargoIDValid(i->produced_cargo[j])) continue;
+			count++;
+		}
+		if (count < 1) {
+			this->DisableWidget(WID_IV_GRAPH);
+		}
```

Comment:
The actual count isn't cared about here, and similar code is used in `ScriptTileList_IndustryProducing::ScriptTileList_IndustryProducing()`, may be worth adding this as a member of Industry in industry.h instead.

```suggestion
		const Industry *i = Industry::Get(window_number);
		if (!i->ProducesCargo()) this->DisableWidget(WID_IV_GRAPH);
```

And I think this should be good for the actual test there:

```c++
	/** Test if this industry produces any cargo.
	 * @return true iff industry produces at least one valid cargo.
	 */
	inline bool ProducesCargo() const
	{
		const auto &found = std::find_if(this->produced_cargo, endof(this->produced_cargo), [](CargoID cid) { return IsCargoIDValid(cid); });
		return found != endof(this->produced_cargo);
	}
```

Reply:
Should I change the code for `ScriptTileList_IndustryProducing::ScriptTileList_IndustryProducing()` in separate commit of this PR too?

Reply:
Yes can do, just explain in the commit title that it is being done to allow re-use of the code.